cmds/core/dirname/dirname.go

```diff
@@ -9,17 +9,27 @@ package main
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"path/filepath"
 )
 
-func main() {
-	if len(os.Args) < 2 {
-		log.Fatalf("dirname: missing operand")
+func runDirname(w io.Writer, args []string) error {
```

Comment:
I would still argue that
runDirname(w io.Writer, dirname string, args ...string) 
as in exec.Command, 
but I can see the case for the way you've done it here. 

I will leave it to you.